AudioPlayer/AudioPlayer/player/AudioPlayerState.swift

```diff
@@ -25,13 +25,13 @@ public enum AudioPlayerError: Error {
 /// - stopped: The player is stopped.
 /// - waitingForConnection: The player is waiting for internet connection.
 /// - failed: An error occured. It contains AVPlayer's error if any.
-public enum AudioPlayerState {
-    case buffering
-    case playing
-    case paused
-    case stopped
-    case waitingForConnection
-    case failed(AudioPlayerError)
+@objc public enum AudioPlayerState: Int {
```

Comment:
Hm, I'd like not to introduce breaking changes for Swift users. What are your thoughts on having a new enum - for example ObjCAudioPlayerState, although we should find a better name - and leave this one for swift users as is?
That means we would have 2 additional properties (one of which is already added - but we'd need to change it)
* `objc_playerState` (again, the naming could be better, it's just to give an example): would convert the swift state into objc state
* `error` would still need to remain a property but could be computed through the current swift state.

The only question that remains with this is how to expose it through a delegate call, but I think it shouldn't be that hard to figure out.

Thoughts on this?